File: 2020/src/test/scala/days/day8Test.scala
package days

import days.day8.extractValue
import org.specs2.mutable.Specification

class day8Test extends Specification {
  val input: Seq[String] =
    fileUtils.loadFile("src/test/scala/days/day8_input.txt")
  val ex1 = List(
    "nop +0",
    "acc +1",
    "jmp +4",
    "acc +3",
    "jmp -3",
    "acc -99",
    "acc +1",
    "jmp -4",
    "acc +6",
  )

  "part 1" should {
    day8.part1(ex1) === 5
    day8.part1(input) === 1446
  }

  "part 2" should {
    day8.part2(ex1) === 8
    day8.part2(input) === 1403
  }
}

File: 2022/src/day8.test.ts
import { part1, part2 } from "./day8";

describe('day 8', () => {
	const testInput = `30373
25512
65332
33549
35390`;

	describe('part 1', () => {
		it('should pass test inputs', () => {
			const cases = [
				{ input: testInput, want: 21 },
			]

			cases.forEach(c => expect(part1(c.input)).toEqual(c.want));
		});
	});

	describe('part 2', () => {
		it.only('should pass test inputs', () => {
			const cases = [
				{ input: testInput, want: 8 },
			]

			cases.forEach(c => expect(part2(c.input)).toEqual(c.want));
		});
	});
});
